src/storage/checkpoint_manager.cpp

```diff
@@ -451,9 +465,18 @@ void CheckpointReader::ReadIndex(ClientContext &context, Deserializer &deseriali
 	}
 
 	D_ASSERT(index_storage_info.IsValid() && !index_storage_info.name.empty());
-	auto art = make_uniq<ART>(info.index_name, info.constraint_type, info.column_ids, TableIOManager::Get(data_table),
-	                          std::move(unbound_expressions), data_table.db, nullptr, index_storage_info);
-	data_table.info->indexes.AddIndex(std::move(art));
+
+	if (info.index_type == "ART") {
+		data_table.info->indexes.AddIndex(make_uniq<ART>(info.index_name, info.constraint_type, info.column_ids,
+		                                                 TableIOManager::Get(data_table), unbound_expressions,
+		                                                 data_table.db, nullptr, index_storage_info));
+	} else {
+		auto unknown_index = make_uniq<UnknownIndex>(info.index_name, info.index_type, info.constraint_type,
+		                                             info.column_ids, TableIOManager::Get(data_table),
+		                                             unbound_expressions, data_table.db, info, index_storage_info);
+
+		data_table.info->indexes.AddIndex(std::move(unknown_index));
+	}
```

Comment:
We can unify this. The ART can still be an unknown index at this point, and we can initialize it the same way we're initializing other indexes later.

Reply:
The ART index type is always setup by the default catalog generator, so AFAIK it can't be uninitialized here.